.github/workflows/doc-build.yml

```diff
@@ -98,28 +99,45 @@ jobs:
         else
           DOC_TARGET="html"
         fi
-
         DOC_TAG=${DOC_TAG} SPHINXOPTS_EXTRA="-q -t publish" make -C doc ${DOC_TARGET}
 
+        # API documentation coverage
+        python3 -m coverxygen --xml-dir  doc/_build/html/doxygen/xml/ --src-dir include/ --output doc-coverage.info
+        # deprecated page causing issues
+        lcov --remove doc-coverage.info \*/deprecated > new.info
+        genhtml --no-function-coverage --no-branch-coverage new.info -o coverage-report
```

Comment:
could you make this a cmake target and add requirements/instructions to the documentation? I think we really want people to be able to run this locally

Reply:
thats what I call feature creep :) 

Can always be done later if there is interest and after the results are optimized, i.e. there are issues with the license boiler plate in many headers being treated as documentation, and probably other issues that we need to deal with before making this official output using cmake. 

Right now this is just about CI and having something to point to and study.

Reply:
> thats what I call feature creep :)

Fair enough  😆

>[...]  there are issues with the license boiler plate in many headers being treated as documentation [...]

No I don't think the headers are the issue. AFAICT it is flagging files that are missing `@file` comment, so the tool says line 1, aka the file, is the culprit.
I am not entirely sure if `@file` is adding value / needed in the first place so we may want to use the `--kind` option to just exclude this check.

Reply:
oh, I was wondering about that, because some files are ok, others are not.

Reply:
:)

It might not be all that useless after all: see https://docs.zephyrproject.org/latest/doxygen/html/files.html especially if you play with the "detail level", it helps depict a better picture of the "physical" code organization, where the defgroups are more logical/semantic grouping. 

The only negative aspect of this is that in many situations `@file`, `@defgroup` name, and `@defgroup` brief end up being pretty much all the exact same text which is not exactly motivating to maintain and get right.